src/autopas/utils/checkFunctorType.h

```diff
@@ -0,0 +1,58 @@
+/**
+ * @file checkFunctorType.h
+ *
+ * @date 28 Aug 2023
+ * @author muehlhaeusser
+ */
+
+#pragma once
+
+#include "autopas/baseFunctors/PairwiseFunctor.h"
+#include "autopas/baseFunctors/TriwiseFunctor.h"
+
+namespace {
+/**
+ * Returns false for any types that are not a PairwiseFunctor
+ * @return false
+ */
+std::false_type isPairwiseFunctorImpl(...);
+/**
+ * Return true for a PairwiseFunctor type
+ * @tparam ParticleT
+ * @tparam FunctorT
+ * @return true
+ */
+template <typename ParticleT, typename FunctorT>
+std::true_type isPairwiseFunctorImpl(autopas::PairwiseFunctor<ParticleT, FunctorT> const volatile &);
+
+/**
+ * Returns false for any types that are not a TriwiseFunctor
+ * @return false
+ */
+std::false_type isTriwiseFunctorImpl(...);
+/**
+ * Return true for a TriwiseFunctor type
+ * @tparam ParticleT
+ * @tparam FunctorT
+ * @return true
+ */
+template <typename ParticleT, typename FunctorT>
+std::true_type isTriwiseFunctorImpl(autopas::TriwiseFunctor<ParticleT, FunctorT> const volatile &);
+}  // namespace
+
+namespace autopas::utils {
+/**
+ * Check whether a Functor Type is inheriting from PairwiseFunctor
+ * @tparam FunctorT
+ */
+template <typename FunctorT>
+using isPairwiseFunctor = decltype(isPairwiseFunctorImpl(std::declval<FunctorT &>()));
```

Comment:
Is the `decltype` needed here and below? (Truly unsure...)

Reply:
Yes it is, though I don't fully grasp why.
GPT says: 
> The `decltype` keyword is essential in this context because it allows the compiler to determine the type of an expression without actually evaluating it. In this particular example, `decltype` is used to deduce the type returned by the `isTriwiseFunctorImpl` function when called with a specific type. This enables the creation of a compile-time constant that reflects whether a given type matches the desired template specialization.

Reply:
Yes, you were right and I changed it to your suggestion.